doc/BMI_MODELS.md

```diff
@@ -0,0 +1,151 @@
+  * implied to be `false` by default
+  
+## BMI Models Written in C
+
+* [Model As Shared Library](#bmi-c-model-as-shared-library)
+* [Example](#bmi-c-cfe-example)
+* [Caveats](#bmi-c-caveats)
+
+### BMI C Model As Shared Library
+
+For **C** models, the model must be packaged as a pre-compiled shared library.  Several CMake cache variables must be configure for controlling whether to expect such a library and how to find it:
+
+* `BMI_C_LIB_ACTIVE` 
+  * type: `BOOL` 
+  * must be set to `ON` (or equivalent in CMake) for BMI C shared library functionality to be compiled and active
+* `BMI_C_LIB_NAME`
+  * type: `STRING `
+  * must be set if `BMI_C_LIB_ACTIVE` is `ON` to supply the appropriate name for the shared library
+* `BMI_C_LIB_HINT_DIR`
+  * type: `STRING`
+  * may be set to provide a [hint](https://cmake.org/cmake/help/latest/command/find_library.html) to CMake when it tries to find the library
+  
+The CMake build system may need to be [regenerated](BUILDS_AND_CMAKE.md#regenerating) after changing these settings.
+
+See the CMake documentation on the *[set](https://cmake.org/cmake/help/latest/command/set.html)* function or [variables](https://cmake.org/cmake/help/latest/manual/cmake-language.7.html#cmake-language-variables) for more information on working with CMake variables. 
+
+When CMake is able to find the library for the given name, it will automatically set up [the dependent, internal, static library](../src/realizations/catchment/CMakeLists.txt) to dynamically link to the external shared library at runtime.  
+
+If `BMI_C_LIB_ACTIVE` is set to `ON`, but either `BMI_C_LIB_NAME` is not set or no library of that name can be found, builds for most (if not all) targets will fail.
+
+See [caveat](#only-one-generic-bmi-c-at-a-time) about only using one **C** BMI model at a time with the built-in realization.
+
+### BMI C CFE Example
+
+An example implementation for an appropriate BMI model as a **C** shared library is provided in the project [here](../extern/cfe).
+
+### BMI C Caveats
+
+* [Activation/Deactivation in CMake Required](#bmi-c-activatedeactivation-required-in-cmake-build)
+* [Only One Generic BMI C at a Time](#only-one-generic-bmi-c-at-a-time)
+* [Additional Bootstrapping Function Needed](#additional-bootstrapping-function-needed)
+
+#### BMI C Activate/Deactivation Required in CMake Build
+
+BMI C functionality will not work (i.e., will not be compiled or executable) unless set to be active in the CMake build.  This requires setting the `BMI_C_LIB_ACTIVE` CMake cache variable to `ON` or `TRUE` (or equivalent).  It will probably also be necessary to configure other settings, as [described here](#bmi-c-model-as-shared-library). 
+
+Conversely, built executables (and perhaps certain build targets) will not work if `BMI_C_LIB_ACTIVE` is `ON` but the configured shared library is not available.
+
+#### Only One Generic BMI C at a Time
```

Comment:
Yeah, this was a side effect of the quick and dirty approach. I wasn't sure how big a deal it would be, or how long it would take to get to, though it also seemed to me like something reasonably easy to work around. Am I off with that?